### PR TITLE
change(jira-auth): change JIRA auth method to use PAT

### PR DESCRIPTION
**Pull request by tomassebestik**
This PR adds support for reading a token from the `JIRA_PASS` variable. This is here due to handle migration process from using username+password to PAT.

- If the `JIRA_PASS` variable starts with the `token:` prefix, token-based authentication will be used (strip `token:` part first).
- If the `JIRA_PASS` variable does not start with a `token:` prefix, basic authentication (username + password) will be used

The example workflows in the README documents retained the current version containing `JIRA_USER`. I recommend to keep it also in the template for new projects. It will be ignored anyway, but since there are already so many projects with it, maybe we can keep it as legacy code.

## Next steps
1) After merging this PR, set on org level:
- `JIRA_USER` (will be ignored during token authentication, but must be present because we don't want to change a ton of existing yaml workflows)
- `JIRA_URL`
- `JIRA_PASS` with token value in the format `token:<token-value>`

2) update Create project tools (remove uploading Jira Secrets to the GH repo)
3) remove `JIRA_USER`, `JIRA_PASS` and `JIRA_URL` from all GH repos by script

**Final state:** 
- all GitHub projects will no longer have `JIRA_USER`, `JIRA_PASS` and `JIRA_URL` secrets, everything will be managed at the org-secrets level

## Related
- https://jira.espressif.com:8443/browse/RDT-567

### CODE REVIEW

**Review by igrr — APPROVED** [review]
Code LGTM, but I haven't tested if it works — I'm assuming you will try this branch in a test repo before merging.

File: sync_issues_to_jira/README.md
Comment (tom-borcin) [inline sync_issues_to_jira/README.md:190]: Shouldn't be?

```suggestion
- `JIRA_PASS` is JIRA token or JIRA password (JIRA basic auth) to log in with
```

Since you are using this env to store token or password in sync_issues_to_jira/sync_to_jira.py

**Comment by tomassebestik** [comment]
> Code LGTM, but I haven't tested if it works — I'm assuming you will try this branch in a test repo before merging.

Thank you @igrr - tested [here](https://github.com/espressif/test-project-bot/actions/runs/6636463569/job/18029093339#step:4:12)
- first run **with JIRA_PASS in the project** -> **authenticated by username+password**, that was expected, test PASSED
- second run J**IRA_PASS removed from the project, org var used** -> **authenticated with token**, expected, test PASSED
- third run, **no secrets in the repo (all goes from org variables)**  -> authenticated with token, expected, **didn't fail due to missing secrets for the workflow**, test PASSED

Seems like our approach is working fine .... I will fix docs as @tom-borcin suggest and merge it.

File: sync_issues_to_jira/README.md
Comment (tomassebestik) [reply sync_issues_to_jira/README.md]: Thank you @tom-borcin , nice catch ... updated!

**Pull request merged**
